cleanlab/dataset.py

```diff
@@ -0,0 +1,411 @@
+    num_examples=None,
+    joint=None,
+    confident_joint=None,
+    multi_label=False,
+):
+    """Prints a health summary of your datasets including results for useful statistics like:
+
+    * The classes with the most and least label issues
+    * Classes that overlap and could potentially be merged
+    * Overall data label quality health score statistics for your dataset
+
+    **Parameters**: For parameter info, see the docstring of :py:func:`find_overlapping_classes <cleanlab.dataset.find_overlapping_classes>`.
+
+    Returns
+    -------
+    dict
+        A dictionary containing keys:
+
+        - ``"health_score"``, corresponding to :py:func:`overall_label_health_score <cleanlab.dataset.overall_label_health_score>`
+        - ``"joint"``, corresponding to :py:func:`estimate_joint <cleanlab.count.estimate_joint>`
+        - ``"classes_by_label_quality"``, corresponding to :py:func:`rank_classes_by_label_quality <cleanlab.dataset.rank_classes_by_label_quality>`
+        - ``"overlapping_classes"``, corresponding to :py:func:`find_overlapping_classes <cleanlab.dataset.find_overlapping_classes>`
+    """
+
+    if joint is None:
+        joint = estimate_joint(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            multi_label=multi_label,
+        )
+    if num_examples is None:
+        num_examples = _get_num_examples(labels=labels)
+
+    print(
+        "*" * 56
+        + "\n"
+        + "Generating a Cleanlab Dataset Health Summary\n"
+        + f" for your dataset with {num_examples:,} examples and {len(joint):,} classes.\n"
+        + "Note, Cleanlab is not a medical doctor... yet.\n"
+        + "*" * 56
+        + "\n",
+    )
+
+    df_class_label_quality = rank_classes_by_label_quality(
+        labels=labels,
+        pred_probs=pred_probs,
+        class_names=class_names,
+        num_examples=num_examples,
+        joint=joint,
+        confident_joint=confident_joint,
+        multi_label=multi_label,
+    )
+    print("Overall Class Quality and Noise across your dataset (below)")
+    print("-" * 60, "\n", flush=True)
+    print(df_class_label_quality)
+
+    df_overlapping_classes = find_overlapping_classes(
+        labels=labels,
+        pred_probs=pred_probs,
+        asymmetric=asymmetric,
+        class_names=class_names,
+        num_examples=num_examples,
+        joint=joint,
+        confident_joint=confident_joint,
+        multi_label=multi_label,
+    )
+    print(
+        "\nClass Overlap. In some cases, you may want to merge classes in the top rows (below)"
+        + "\n"
+        + "-" * 83
+        + "\n",
+        flush=True,
+    )
+    print(df_overlapping_classes)
+    print()
+    health_score = overall_label_health_score(labels=labels, pred_probs=pred_probs, verbose=True)
+    print("\nGenerated with <3 from Cleanlab.")
+    return {
+        "health_score": health_score,
+        "joint": estimate_joint,
```

Comment:
BUG:  This "joint" key was a function for me in my test run:

<function estimate_joint at 0x1236dc4c0>

Does not seem right

Reply:
i think you fixed this. thanks